packages/ui/src/SegmentedPicker/index.tsx

```diff
@@ -0,0 +1,137 @@
+
+const outlineColorByActionType: Record<ActionType, keyof DefaultTheme['color']['action']> = {
+  default: 'neutralFocusOutline',
+  accent: 'primaryFocusOutline',
+  unshield: 'unshieldFocusOutline',
+};
+
+const SegmentButton = styled.button<{
+  $actionType: ActionType;
+  $getFocusOutlineColor: (theme: DefaultTheme) => string;
+  $getBorderRadius: (theme: DefaultTheme) => string;
+}>`
+  flex-grow: 1;
+  flex-shrink: 1;
+  flex-basis: 0; /** Ensure equal widths */
+
+  appearance: none;
+  background-color: transparent;
+  border: none;
+  border-radius: ${props => props.theme.borderRadius.xs};
+  color: ${props => {
+    switch (props.$actionType) {
+      case 'accent':
+        return props.theme.color.primary.light;
+      case 'unshield':
+        return props.theme.color.unshield.light;
+      default:
+        return props.theme.color.text.primary;
+    }
+  }};
+  position: relative;
+  white-space: nowrap;
+  cursor: pointer;
+
+  ${tab}
+  ${buttonInteractions}
+
+  &::after {
+    inset: ${props => props.theme.spacing(0.5)};
+  }
+`;
+
+const SelectedIndicator = styled(motion.div)`
+  background-color: ${props => props.theme.color.text.primary + TEN_PERCENT_OPACITY_IN_HEX};
+  border-radius: ${props => props.theme.borderRadius.xs};
+  position: absolute;
+  inset: 0;
+  z-index: -1;
+`;
+
+export interface SegmentedPickerOption<ValueType> {
+  /**
+   * The value to pass to the `onChange` handler when clicked. Must be unique
+   * across all segments, and must be either a string, number, or an object with
+   * a `.toString()` method so that it can be used as a React key.
+   */
+  value: ValueType;
+  label: string;
+  disabled?: boolean;
+}
+
+export interface SegmentedPickerProps<ValueType extends { toString: () => string }> {
+  /**
+   * The currently selected value. Will be compared to the `options`' `value`
+   * property using `===` to determine which segment is selected.
+   */
+  value: ValueType;
+  onChange: (value: ValueType) => void;
+  options: SegmentedPickerOption<ValueType>[];
+  actionType?: ActionType;
+}
+
+/**
+ * Renders a segmented picker where only one option can be selected at a time.
+ * Functionally equivalent to a `<select>` element or a set of radio buttons,
+ * but looks nicer when you only have a few options to choose from. (Probably
+ * shouldn't be used with more than 5 options.)
+ *
+ * @example
+ * ```TSX
+ * <SegmentedPicker
+ *   value={value}
+ *   onChange={setValue}
+ *   options={[
+ *     { value: 'one', label: 'One' },
+ *     { value: 'two', label: 'Two' },
+ *     { value: 'three', label: 'Three', disabled: true },
+ *   ]}
+ * />
+ * ```
+ */
+export const SegmentedPicker = <ValueType extends { toString: () => string }>({
+  value,
+  onChange,
+  options,
+  actionType = 'default',
+}: SegmentedPickerProps<ValueType>) => {
+  const layoutId = useId();
+
+  return (
+    <Root>
+      {options.map(option => (
+        <SegmentButton
```

Comment:
thought/issue (a11y): semantically, `SegmentedPicker` is used as either a radio button group, or as tabs. Each of these usages have their W3C standards for accessibility. For now, the component doesn't fill the standards fully:
1. Tabs: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/. `SegmentedPicker` doesn't support arrow navigation and aria attributes like `aria-selected` for the selected button, etc.
2. Radio group: https://www.w3.org/WAI/ARIA/apg/patterns/radio/. `SegmentedPicker` doesn't use the required roles and keyboard navigation.

I don't mind having this version of a component for now – it looks good and works correctly, at least partially accessible with the keyboard. Yet, I believe that in the future `SegmentedPicker` should be split into `RadioGroup` and `Tabs` components correspondingly.

Reply:
Ah, this is a good call-out — this seems like a great candidate for using an already existing component (like Radix UI's [Tabs](https://www.radix-ui.com/primitives/docs/components/tabs)). Since we're going to be using `<SegmentedPicker />` for tabs anyway, that seems like the right fit. I'll move this PR back to draft until this is addressed.

Reply:
Actually, on second thought, I'll do this as a separate PR to keep PRs smaller.